test/cpp/common/alarm_test.cc

```diff
@@ -438,6 +439,34 @@ TEST(AlarmTest, UnsetDestruction) {
   Alarm alarm;
 }
 
+TEST(AlarmTest, CallbackSetInCallback) {
+  auto c = std::make_shared<Completion>();
+  struct GuardedAlarm {
+    grpc::internal::Mutex mu;
```

Comment:
Instead of relying on gRPC internals for a test that simulates public usage, try `std::mutex`. It's being used here already (see 10 lines down).

Reply:
done.